### DISCUSSION TITLE
Compatibility between ArgoCD and Argo rollouts

### DISCUSSION THREAD

**Discussion by javiermmenendez** [argoproj/argo-rollouts#1590] [Ideas]
We have been using ArgoCD very successfully in our Openshift installation and we want to keep it that way. Just days ago we started testing Argo Rollouts and we liked it very much. We think that it can be a very useful tool. But we have observed what we think could be a problem when we try to join both tools ArgoCD and Argo Rollouts. 
When we define in our Git ArgoCD project Rollout and Service objects, due to the way that Argo Rollout works, when it need to modify then it results in errors and sync issues. For example with services we get this message:

Error updating Endpoint Slices for Service pcr-pro/stable-service: failed to update stable-service-mtw59 EndpointSlice for Service pcr-pro/stable-service: Operation cannot be fulfilled on endpointslices.discovery.k8s.io "stable-service-mtw59": the object has been modified; please apply your changes to the latest version and try again

We think that it could be possible to have an upper object that creates rollout and service objects instead to creating them manually.

Thanks in advance.

**Comment by harikrongali** [comment]
@javiermmenendez I am not familiar with openshift but we have been using Argo rollouts + ArgoCD together at Intuit successfully at scale. I might require more info regarding what do you mean by upper object creating rollout & service objects?

**Comment by javiermmenendez** [comment]
You are right. All elements keep synchorinized. We just started with Argo Rollouts and all was due to configuration mistakes . We think we will keep testing, and is likely that we will use this tool.

Sorry and thanks
